### DISCUSSION TITLE
socket.io client question

### DISCUSSION THREAD

**Discussion by zkcrescent** [zishang520/socket.io#57] [Q&A]
from official documents, web side socket.io client can connect to special namespace, how to handle the connection event on server side ? 

client side:
``` js 

import { Socket, io } from "socket.io-client"

const socket = io("/admin")

```

server side:
``` golang 

skio := socket.NewServer( types.NewWebServer(nil), nil)

skio.On("/admin", func(clis ...any){})

```
not works.

**Comment by zkcrescent** [comment]
when useing 
``` golang
skio := socket.NewServer( types.NewWebServer(nil), nil)

skio.On("connection", func(clis ...any){})
```
it does not catch the `io("/admin")` connection event

**Comment by zishang520** [comment]
By default, `skio` listens to the root (`/`) event. You need to listen to the `connection` event after `Of`:

```golang
skio.Of("/admin", nil).On("connection", func(clis ...any){})
```

**Comment by zishang520** [comment]
If you need dynamic namespace you can use regexp expression:

```golang
skio.Of(regexp.MustCompile(`/\w+`), nil).On("connection", func(clis ...any){})
```

**Comment by zkcrescent** [comment]
thanks for your help. Another question is that I want to wrap (echo)[github.com/labstack/echo/v4] context  when on connection, how can I do for this.

**Reply by zishang520** [reply]
Can you briefly describe your business scenario? Or a pseudo code, I will try my best to give you an answer based on your needs.
